draft-ietf-tls-tls13.md

```diff
@@ -3025,17 +2937,263 @@ processing the server's Certificate, CertificateVerify, and
+  in the case of server authentication or client authentication
+  in 0-RTT, this field SHALL be zero length.
+
+certificate_list
+: This is a sequence (chain) of certificates. The sender's
+  certificate MUST come first in the list. Each following
+  certificate SHOULD directly certify one preceding it. Because
+  certificate validation requires that trust anchors be distributed
+  independently, a certificate that specifies a
+  trust anchor MAY be omitted from the chain, provided that
+  supported peers are known to possess any omitted certificates.
+{:br }
+
+Note: Prior to TLS 1.3, "certificate_list" ordering required each certificate
+to certify the one immediately preceding it,
+however some implementations allowed some flexibility. Servers sometimes send
+both a current and deprecated intermediate for transitional purposes, and others
+are simply configured incorrectly, but these cases can nonetheless be validated
+properly. For maximum compatibility, all implementations SHOULD be prepared to
+handle potentially extraneous certificates and arbitrary orderings from any TLS
+version, with the exception of the end-entity certificate which MUST be first.
+
+The server's certificate list MUST always be non-empty. A client MAY
+MAY send no certificates if it does not have an appropriate
+certificate to send in response to the server's authentication
+request.
+
+##### Server Certificate Selection
+
+The following rules apply to the certificates sent by the server:
+
+-  The certificate type MUST be X.509v3 {{RFC5280}}, unless explicitly negotiated
+  otherwise (e.g., {{RFC5081}}).
+
+-  The server's end-entity certificate's public key (and associated
+  restrictions) MUST be compatible with the selected key exchange
+  algorithm.
+
+~~~~
+    Key Exchange Alg.  Certificate Key Type
+
+    DHE_RSA            RSA public key; the certificate MUST allow the
+    ECDHE_RSA          key to be used for signing (i.e., the
+                       digitalSignature bit MUST be set if the key
+                       usage extension is present) with the signature
+                       scheme and hash algorithm that will be employed
+                       in the server's KeyShare extension.
+                       Note: ECDHE_RSA is defined in [RFC4492].
+
+    ECDHE_ECDSA        ECDSA or EdDSA public key; the certificate
+                       MUST allow the key to be used for signing with the
+                       allow the key to be used for signing with the
+                       hash algorithm that will be employed in the
+                       server's KeyShare extension.  The public key
+                       MUST use a curve and point format supported by
+                       the client, as described in [RFC4492].
+~~~~
+
+- The "server_name" and "trusted_ca_keys" extensions {{RFC6066}} are used to
+  guide certificate selection. As servers MAY require the presence of the server_name
```

Comment:
"server_name" has quotes around it above, but not here; we've generally been quoting around all extension names, so this needs it too